src/test/clojure/xtdb/operator/scan_test.clj

```diff
@@ -694,42 +694,95 @@
                                         :for-valid-time :all-time})))))))
 
 (deftest test-recency-filtering
-  (tu/with-tmp-dirs #{node-dir}
-    (binding [c/*page-size* 1
-              c/*l1-file-size-rows* 256
-              c/*ignore-signal-block?* true]
-      (with-open [node (xtn/start-node {:log [:in-memory {:instant-src (tu/->mock-clock (tu/->instants :year))}]
-                                        :storage [:local {:path (io/file (.toFile node-dir) "objects")}]
-                                        :indexer {:rows-per-chunk 1}})]
-        ;; 2020 - 2025
-        (let [tx-key (last (for [i (range 6)]
-                             (xt/execute-tx node [[:put-docs :docs {:xt/id 1 :col i}]])))]
+  (binding [c/*page-size* 1
```

Comment:
This is a good high-level test, definitely worth keeping - but part of the aim of our refactoring earlier in the week was that we can then easier unit-test the lower-level function. With the new tests as they stand, we don't necessarily know that we've filtered the data out at the page level?

Reply:
aha - they're in trie-test. worth bringing those tests into this namespace, given they're testing scan functions?

Reply:
Move `->merge-task` to `trie.clj`